jonfroehlich/makeabilitylabwebsite
# Transitioning to postgres on the dev environment

### Overview
This pull request is an initial step to addressing issue #366.

Per @jonfroehlich's suggestion, this pull request transitions from using db.sqlite3 in the dev environment to using the Postgres container in Docker.

Docker-compose.yml file is based off of the one from https://wsvincent.com/django-docker-postgresql/ with a persistent volume mount added to make sure the database persists between containers.

### Changes
1. Added a docker-compose.yml to run the makeability website container in tandem with the Postgres container
2. Changed the Dockerfile to match more standard convention (using COPY instead of ADD)
     - Note: the removal of the `EXPOSE 8000` command is to remove redundancy, as the docker-compose file will do the exact same thing
3. Changes to the main settings.py include:
     -  Changing the `DATABASES `section to connect to the Postgres Docker container
     - Changing the `ALLOWED_HOSTS` to allow all on the dev environment

### Testing
To test this PR, simply pull this branch and run the following commands:
1. Run `docker build . -t makelab_image` to install all dependencies (Note: this could take a while)
2. In order to log into the django admin interface, run `docker-compose run web python /code/manage.py createsuperuser`
3. To verify that the website is working, run `docker-compose up` and navigate to localhost:8000 
4. Run `docker-compose down` to shut down the containers

Note: beyond this point you can run `docker-compose up -d` to remove the logging messages that Docker outputs

5. Make changes to the website/models.py file and run `docker-compose up` (Docker will auto-build and make migrations)
6. When you have verified that changes have been made properly, shut down the system with `docker-compose down` again.
